### PR TITLE
Add mask width and gradient color width attribute

### PR DESCRIPTION
**Pull request by veghtomi**
Introduces feature mentioned in #26.

### CODE REVIEW

File: shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java [hunk 1]
```diff
@@ -171,6 +177,44 @@ public void setShimmerAngle(int angle) {
         resetIfStarted();
     }
 
+    /**
+     * Sets the width of the shimmer line to a value higher than 0 to less or equal to 1.
+     * 1 means the width of the shimmer line is equal to the half of the width of the ShimmerLayout.
```
Comment (szugyi) [inline shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java]: > equal to half of the width

File: shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java [hunk 2]
```diff
@@ -171,6 +177,44 @@ public void setShimmerAngle(int angle) {
...
+     * This value must be higher than {@link #gradientCenterColorWidth} or the shape of the shimmer line
+     * will not look like as expected.
+     * The default value is 0.5.
+     *
+     * @param maskWidth The width of the shimmer line.
+     */
+    public void setMaskWidth(float maskWidth) {
+        if (maskWidth <= 0 || 1 < maskWidth) {
+            throw new IllegalArgumentException(String.format("maskWidth value must be higher than %d and less or equal to %d",
+                    0, 1));
```
Comment (szugyi) [inline shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java]: If these are format parameters, you could extract them as constants, and use them in the if condition too, to make sure they are the same

File: shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java [hunk 3]
```diff
@@ -171,6 +177,44 @@ public void setShimmerAngle(int angle) {
...
+        }
+
+        this.maskWidth = maskWidth;
+        resetIfStarted();
+    }
+
+    /**
+     * Sets the width of the center gradient color to a value higher than 0 to less or equal to 1.
+     * 1 means that the whole shimmer line will has this color without transparent edges.
```
Comment (szugyi) [inline shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java]: > will have

File: shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java [hunk 4]
```diff
@@ -171,6 +177,44 @@ public void setShimmerAngle(int angle) {
...
+     * This value must be less than {@link #maskWidth} or the shape of the shimmer line
+     * will not look like as expected.
+     * The default value is 0.06.
+     *
+     * @param gradientCenterColorWidth The width of the center gradient color.
+     */
+    public void setGradientCenterColorWidth(float gradientCenterColorWidth) {
+        if (gradientCenterColorWidth <= 0 || 1 < gradientCenterColorWidth) {
+            throw new IllegalArgumentException(String.format("gradientCenterColorWidth value must be higher than %d and less or equal to %d",
+                    0, 1));
```
Comment (szugyi) [inline shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java]: Same as for `setMaskWith`

**Review by szugyi — CHANGES_REQUESTED** [review]

File: shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java (hunk 1 above)
Comment (veghtomi) [reply shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java]: Fixed.

File: shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java (hunk 2 above)
Comment (veghtomi) [reply shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java]: Fixed.

File: shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java (hunk 3 above)
Comment (veghtomi) [reply shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java]: Fixed.

File: shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java (hunk 4 above)
Comment (veghtomi) [reply shimmerlayout/src/main/java/io/supercharge/shimmerlayout/ShimmerLayout.java]: Fixed.

**Pull request merged**